### ISSUE TITLE
[WhatsApp] suddenly stopped connecting, and SMS verification fails

### ISSUE THREAD

**Issue by 03927e9c** [loqui/im#872]
WhatsApp stopped connecting (green-yellow switch), it's stucked in yellow. Uninstalled and reinstalled latest version and can't receive SMS.

**Comment by TitanNano** [comment]
@cmeerw can you get a new token? I'm not sure whats going on here... is our version and token outdated?

**Comment by cmeerw** [comment]
I don't know what's going on either, the token should be up to date...

**Comment by TitanNano** [comment]
really strange. I just get an "unauthorized" error...

**Comment by jcerdan** [comment]
Hi, I have exactly the same problem... tried to get a new code but always get unauthorized

**Comment by Merkur0815** [comment]
Hi, same problem here a few hours ago...

**Comment by TitanNano** [comment]
thanks to everyone. This seems to be a general problem with the WhatsApp server. Right now I can't say whats the problem.

**Comment by jcerdan** [comment]
Hi,
I have created an account with yowsup on Pidgin and it's working fine. Had to modify Resource and set it to: "S 40-2.12.11-433" because of encryption issues. (known since september)
That's a solution while this issue is fixed...

**Comment by cmeerw** [comment]
Cool, using "S 40-2.12.11-443" as the resource actually works with
Loqui, thanks for the hint.

**Comment by TitanNano** [comment]
oh yes! but why "S 40" and not "S40"?? 😅

**Comment by TitanNano** [comment]
@cmeerw  I'll create a patch.

**Comment by divyaadupala** [comment]
Hii
iam using moqui im watsapp suddenly stopped it does not changing yellow to green uniinstall and reinstall again it was not getting any code and showing athenucating error

**Comment by TitanNano** [comment]
is one of you able to create a build and test if you can get a code?

**Comment by cmeerw** [comment]
I can test and build in the evening.

BTW, I am not sure if "v", "u" and "d" should change as well or if
those should remain as they are now...

**Comment by TitanNano** [comment]
yes, I'm also not sure.

**Comment by TitanNano** [comment]
we can't keep this resource string...
![2015-11-05-14-46-49](https://cloud.githubusercontent.com/assets/4500858/10972476/b46a61cc-83d9-11e5-82c2-16f2b7081888.png)

**Comment by cmeerw** [comment]
What did you do to get that error? (I haven't seen that on my phone
with that resource string...)

**Comment by TitanNano** [comment]
with the new resource string... after I connected the second time i think. I had to restart loqui and received the message again.

**Comment by Zeus76** [comment]
Hello, yes the code they send you, but when you try to sign up for Whatsapp said authentication error (the image is in Spanish).
Reads: "status", "fail", "reason", "incorrect", "sms_length": 6, "voice_length": 6
![whatsapp](https://cloud.githubusercontent.com/assets/15674458/10974294/b304ddc8-83e1-11e5-9f4d-8174359481f5.png)

**Comment by TitanNano** [comment]
@cmeerw hmm strange. now I flashed a new build with d also changed and now I don't receive the message any more..

@Zeus76 but after that it says status ok, you get your password and the stream starts...

**Comment by Zeus76** [comment]
I get is the verification message 6 numbers that are sent by SMS Whatsapp. But when I put that number of 6 numbers to register, it makes me "authentication error"

**Comment by TitanNano** [comment]
@Zeus76 but the log you posted shows that you got your password and loqui starts to connect. I don't understand what exactly your problem is.

**Comment by Zeus76** [comment]
I get is the verification message 6 numbers that are sent by SMS Whatsapp. But when I put that number of 6 numbers to register, it makes me "authentication error"
Loqui fails to connect, "Authentication failed" comes and goes and stays in the registration screen Whatsapp. Fails to connect.
In the simulator appears: "status":"fail", "reason":"incorrect","sms_length": 6, "voice_length": 6

**Comment by TitanNano** [comment]
did you use a build from master or from the marketplace?

**Comment by Zeus76** [comment]
From the Marketplace.

**Comment by TitanNano** [comment]
@Zeus76 okay, this is what this issue is about. we're trying to fix the problem and release a new version as soon as possible.

**Comment by marcfp** [comment]
Hi people,
I'm using loqui from marketplace and i can't connect too...
I'm using firefox os 1.3 into alcatel one touch fire C

**Comment by TitanNano** [comment]
@marcfp yes the marketplace version doesn't work anymore. We are working on an update.

**Comment by Boersenfeger** [comment]
Hi, unfortunately, the MarketPlace-App is also currently not working.. After the Loquoi-Update has completed, we could install it using the website? For this purpose, then the current version must be uninstalled? Sorry, my English is bad..

**Comment by TitanNano** [comment]
@Boersenfeger Please simply wait for an update. It will come as soon as possible. You don't have to uninstall anything.

**Comment by Boersenfeger** [comment]
OK, Thanks a lot..

**Comment by tibtab** [comment]
I also received  "Unable to parse recource" after building the newest version, but only once. And besides that anything is ok.

**Comment by cmeerw** [comment]
But registration fails with "old_version" with that tokenData. So I am thinking of reverting back to 2.13.9 and only changing "S40" to "S 40" in "r" and "d".

**Comment by cmeerw** [comment]
Hmm, actually not sure now... yowsup appears to be able to connect with "S40"...

**Comment by TitanNano** [comment]
O.o this is really strange...

**Comment by Aelerinya-old** [comment]
For me too the marketplace app doesn't work anymore.

The output in WebIDE :
![log](http://image.noelshack.com/fichiers/2015/45/1446751944-capture-du-2015-11-05-20-28-06.png)

**Comment by cmeerw** [comment]
Please test with commit 0f6493cb0c389bf135aec7a8892800c61fcd7dd5 (and review the readNibble function)

**Comment by TitanNano** [comment]
work for me :+1:

**Comment by tibtab** [comment]
loqui works fine for me, but how shall I notice if this read Nibble thing works? What is it for?

**Comment by cmeerw** [comment]
@tibtab I guess if WhatsApp works for you then there is a good chance that readNibble also works :) I guess this is now mainly used to encode phone numbers in the protocol stream.

**Comment by tibtab** [comment]
ok :)

**Comment by tibtab** [comment]
There is one little thing which is strange. There are no acks displayed for me any more exept for in the preview. It's really strange, the first ack appears and then suddenly disappears.

**Comment by FabianOvrWrt** [comment]
It doesn't work for me. I downloaded 0.4.9 and it gets stuck on "Waiting for SMS..." and then it just gets back to the registration screen.

The WebIDE just shows "[l10n] #contactAdd is undefined".

**Comment by divyaadupala** [comment]
I have the same problem after entering the verification code it was showing authentication failed error

**Comment by TitanNano** [comment]
@tibtab it's an style issue. I'll land the patch today.

**Comment by Boersenfeger** [comment]
Where can I download the version 0.4.9 or comes this "over the Fly"?

**Comment by TitanNano** [comment]
you're phone will tell you about the update as soon as the marketplace reviewers approved it.

**Comment by Boersenfeger** [comment]
Thanks TitanNano.. :-))

**Comment by marcfp** [comment]
Marketplace doesn't run in my ffox os 1.3 and loqui neither run

**Comment by cmeerw** [comment]
0.4.9 is now available in the marketplace...

**Comment by 03927e9c** [comment]
Now I don't get an error but the SMS or phone call don't arrive.

**Comment by 03927e9c** [comment]
Someone tried the latest version from MK? Is it working for you?
EDIT: In the app comments people are having the same problem (in spanish):
![pantallazo-ventana sin titulo](https://cloud.githubusercontent.com/assets/9679446/11012381/ebfaf914-84d1-11e5-8166-e2a48a6c7f30.png)

**Comment by TitanNano** [comment]
I will check it today and see what's the issue here.

**Comment by Boersenfeger** [comment]
The same Failure here! Im get no Update, so I deinstall Loqui IM, go to FFOS Marketplace, download new Version, started Loqui IM for registration procedure. But no SMS arrive... :-((

**Comment by cmeerw** [comment]
Ok, sorry - seeing it now as well - I believe this is fixed with #876 - not sure how I missed that when I tested it on Thursday (well, I probably hadn't merged all the changes when testing registration)

**Comment by TitanNano** [comment]
@cmeerw why was that a problem?

**Comment by cmeerw** [comment]
@TitanNano I got console not defined on line 3546 - and I believe it's related to some weird JavaScript scoping rules... (btw, you only see the problem when you start Loqui from scratch with no accounts and try to register a new account)

**Comment by TitanNano** [comment]
@cmeerw oh okay I see.

**Comment by Boersenfeger** [comment]
... and now we have to wait for the reviewers OK!?

**Comment by cmeerw** [comment]
We have to upload another version to the marketplace, unfortunately...

@TitanNano you mentioned a fix for the "ack ticks" style issue - is that a simple one we can include as well?

**Comment by TitanNano** [comment]
@cmeerw yes! I will push it.

**Comment by Boersenfeger** [comment]
Thank you for the quickly Response... :+1:

**Comment by cmeerw** [comment]
Hopefully won't be too long for the fix to be available in the marketplace...

**Comment by Boersenfeger** [comment]
mee tooo  :-))

**Comment by suoko** [comment]
I built the latest working one and put it here https://www.dropbox.com/s/h5gufnndmg1u82a/loquiim_firefoxos.tar.gz?dl=0
Push it with webide

**Comment by BlackForest2015** [comment]
Thank's suoko, runs perfect!

**Comment by suoko** [comment]
On mine it can send messages but can't receive them.
Also, whenever I uninstall and reinstall, it does not send a new reigistration sms, it just connects. I'm not sure it's completely uninstalled.
Where can I look for app leftovers ?

**Comment by TitanNano** [comment]
uninstalling Loqui will remove all application files and related userdata. But it won't change anything. 
I'm also not able to reproduce your issue, I have no problems to receive messages. Please file a new issue for your problem.

**Comment by suoko** [comment]
You know what, Something like one week  ago I was experiencing this problem: I could receive from and send messages to groups but not from / to single buddies.
So I reinstalled the app from marketplace and Everything worked again but.... Only till the following day when it broke for everybody causing this issue to be created.
Now I noticed I'm experiencing what follows:
With groups no problem, with single buddies I can send everything but I can receive only images, not text.

**Comment by 03927e9c** [comment]
Latest version from Market Place is now working.

**Comment by TitanNano** [comment]
@suoko please file a new issue and we will investigate it.

**Comment by suoko** [comment]
It must be something related to my whatsapp account because that happens even on the firefoxos emulator. I should actually talk to whatsapp (maybe via telegram :+1:  )

**Comment by Boersenfeger** [comment]
Thank you very much, no more Problems here :+1:

**Comment by luiguild** [comment]
This problem again?

**Comment by luiguild** [comment]
![captura de tela 2016-02-11 as 00 27 40](https://cloud.githubusercontent.com/assets/16566338/12968270/8c562698-d056-11e5-9d36-b3f93cf743fe.jpg)
I can't connect into my whatsapp account (BR) =(
